content-service.jenkinsfile

```diff
@@ -0,0 +1,94 @@
+pipeline {
+    agent any
+    environment
+    {
+      // NOTE: this file must be executed in a directory whose name is a numeric sequence, and whose parent is named "Releases".
+      // This is how other Jenkinsfiles in the Release process determine the current release number.
+      RELEASE_VERSION = (pwd() =~ /Releases\/(\d+)\//)[0][1];
+    }
+
+    stages
+    {
+        stage('Checkout')
+        {
+            steps
+            {
+                // Get some code from a GitHub repository
+                git branch: 'feature/Joomla', url: 'https://github.com/reactome/container.git'
+            }
+        }
+        stage('Download graphdb')
+        {
+            steps
+            {
+                fileOperations([fileDownloadOperation(password: '', proxyHost: '', proxyPort: '', targetFileName: "reactome-${env.RELEASE_VERSION}.graphdb.tgz", targetLocation: './neo4j', url: 'https://reactome.org/download/current/reactome.graphdb.tgz', userName: '')])
+            }
+        }
+
+        stage("Build graphdb")
+        {
+            steps
+            {
+                dir("neo4j")
+                {
+                    script
+                    {
+                        docker.build("reactome/graphdb:$RELEASE_VERSION --build-arg GRAPHDB_LOCATION=./reactome-${env.RELEASE_VERSION}.graphdb.tgz --build-arg RELEASE_VERSION=${env.RELEASE_VERSION} -f ./neo4j_stand-alone.dockerfile")
+                    }
+                }
+            }
```

Comment:
For steps like this that are common to all 3 Jenkins files, is it possible to create a separate file in Jenkins that acts as a module for common functionality that you can import and re-use?

Reply:
I know we've asked @cookersjs about this but from what I can tell, he hasn't found a solution to this problem either. I agree that this would be nice, but I think that can wait for a later iteration, if it's possible.

Reply:
Okay - it's not a blocker.